ndb_api/db_helpers.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2022-2023 Nutanix, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ndb_api
+
+import (
+	"context"
+	"errors"
+	"strconv"
+
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// This function generates and returns a request for provisioning a database (and a dbserver vm) on NDB
+// The database provisioned has a NONE time machine SLA attached to it, and uses the default OOB profiles
+func GenerateProvisioningRequest(ctx context.Context, ndb_client *ndb_client.NDBClient, database DatabaseInterface, reqData map[string]interface{}) (requestBody *DatabaseProvisionRequest, err error) {
+	log := ctrllog.FromContext(ctx)
+	log.Info("Entered ndb_api_helpers.GenerateProvisioningRequest", "database name", database.GetDBInstanceName(), "database type", database.GetDBInstanceType())
+
+	// Fetching the NONE TM SLA
+	sla, err := GetNoneTimeMachineSLA(ctx, ndb_client)
+	if err != nil {
+		log.Error(err, "Error occurred while getting NONE TM SLA", "database name", database.GetDBInstanceName(), "database type", database.GetDBInstanceType())
+		return
```

Comment:
dont you need to return 2 things from this function? requestBody and err?

Reply:
Yes, we do.

The return here is called a 'naked' return, since we have 'named' return values, a naked return would return those named values. 
The point of adding the named return values was to avoid instantiation (and initialization) across functions repeatedly since these are zero initialized be default. 
This is a pattern that has been followed quite extensively in the codebase. Let me know if we want explicit returns?
Ref: https://go.dev/tour/basics/7

Reply:
I guess @manavrajvanshi is doing something called as a "naked return in golang" (which is also something I learned last week :D )